app/furniture/marketplace/delivery_area_component.rb

```diff
@@ -28,20 +28,30 @@ def edit_button?
       delivery_area.persisted? && policy(delivery_area).edit?
     end
 
+    def discard_button
+      return unless discard_button?
+
+      ButtonComponent.new(label: "#{t("icons.discard")} #{t("discard.link_to")}",
+        title: t("marketplace.delivery_areas.discard.link_to", name: delivery_area.label),
+        href: delivery_area.location, method: :delete, scheme: :secondary)
+    end
+
+    def discard_button?
+      delivery_area.persisted? && !delivery_area.discarded? && policy(delivery_area).destroy?
```

Comment:
`!discarded?` is the same as `kept?` and also the same as `undiscarded?`, I would slightly prefer:
```suggestion
      delivery_area.persisted? && delivery_area.kept? && policy(delivery_area).destroy?
```